deploy/001_deploy_InverseApi3ReaderProxyV1.ts

```diff
@@ -0,0 +1,58 @@
+import type { HardhatRuntimeEnvironment } from 'hardhat/types';
+
+import { getDeploymentName } from '../src';
+
+export const CONTRACT_NAME = 'InverseApi3ReaderProxyV1';
+
+module.exports = async (hre: HardhatRuntimeEnvironment) => {
+  const { getUnnamedAccounts, deployments, ethers, network, run } = hre;
+  const { deploy, log } = deployments;
+
+  const [deployerAddress] = await getUnnamedAccounts();
+  if (!deployerAddress) {
+    throw new Error('No deployer address found.');
+  }
+  log(`Deployer address: ${deployerAddress}`);
+
+  const proxyAddress = process.env.PROXY;
+  if (!proxyAddress) {
+    throw new Error('PROXY environment variable not set. Please provide the address of the proxy contract.');
+  }
+  if (!ethers.isAddress(proxyAddress)) {
+    throw new Error(`Invalid address provided for PROXY: ${proxyAddress}`);
+  }
+  log(`Proxy address: ${proxyAddress}`);
+
+  const isLocalNetwork = network.name === 'hardhat' || network.name === 'localhost';
+
+  const confirmations = isLocalNetwork ? 1 : 5;
```

Comment:
Isn't 5 a bit too much?

Reply:
On some chains 5 confirmations might take quite a long time but on most it won't. This number was suggested to me by the hardhat-deploy plugin when deploying against ethereum-sepolia-testnet and usually 5 confirmations is a common safety measure for non-local deployments, balancing speed with protection against blockchain reorganizations.